jdaviz/configs/mosviz/tests/test_parsers.py

```diff
@@ -1,48 +1,96 @@
 from zipfile import ZipFile
-import pathlib
+from tempfile import TemporaryDirectory
+from pathlib import Path
 
 import pytest
 from astropy.utils.data import download_file
 
 from jdaviz.utils import PRIHDR_KEY, COMMENTCARD_KEY
 
 
-# This is another version of test_niriss_loader in test_data_loading.py
 @pytest.mark.remote_data
+@pytest.mark.filterwarnings('ignore', match="'(MJy/sr)^2' did not parse as fits unit")
 def test_niriss_parser(mosviz_helper, tmpdir):
+    '''
+    Tests loading a NIRISS dataset
+    This data set is a shortened version of the ERS program GLASS (Program 1324)
+    provided by Camilla Pacifici. This is in-flight, "real" JWST data
 
-    test_data = 'https://stsci.box.com/shared/static/l2azhcqd3tvzhybdlpx2j2qlutkaro3z.zip'
-    fn = download_file(test_data, cache=True, timeout=30)
-    with ZipFile(fn, 'r') as sample_data_zip:
-        sample_data_zip.extractall(tmpdir)
+    The spectra are jw01324001001_15101_00001_nis
+    The direct image is jw01324-o001_t001_niriss_clear-f200w
+    Please see JWST naming conventions for the above
 
-    level3_path = (pathlib.Path(tmpdir) / 'NIRISS_for_parser_p0171')
+    The dataset was uploaded to box by Duy Nguyen
+    '''
 
-    data_dir = level3_path
+    # Download data
+    test_data = 'https://stsci.box.com/shared/static/cr14xijcg572dglacochctr1kblsr89a.zip'
+    fn = download_file(test_data, cache=True, timeout=30)
 
-    mosviz_helper.load_niriss_data(data_dir)
+    # Extract to a known, temporary folder
+    data_dir = Path(TemporaryDirectory(dir=tmpdir).name)
```

Comment:
The point of `tmpdir` is that you do not have to create temporary directory yourself. But at this point, might as well use `tmp_path` fixture. See https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

Reply:
> provide a temporary directory unique to the test invocation

Ahh, I didn't know it was unique to that specific test; great! I'll fix that